framework/platform/bpl/uci/db/bpl_db_uci.cpp

```diff
@@ -95,6 +95,44 @@ bool uci_add_section(const std::string &package_name, const std::string &section
     LOG(TRACE) << "uci_add_section() " << package_name << ".(" << section_type << ")"
                << section_name;
 
+    auto ctx = alloc_context();
+    if (!ctx) {
+        return false;
+    }
+
+    char sec_path[MAX_UCI_BUF_LEN] = {0};
+    if (snprintf(sec_path, MAX_UCI_BUF_LEN, section_type_path, package_name.c_str(),
+                 section_name.c_str(), section_type.c_str()) <= 0) {
+        LOG(ERROR) << "Failed to compose path";
+        return false;
+    }
+
+    uci_ptr ptr;
+    // Get pointer to value
+    if (uci_lookup_ptr(ctx.get(), &ptr, sec_path, true) != UCI_OK) {
+        LOG(ERROR) << "UCI failed to lookup ptr for path: " << sec_path << std::endl
+                   << uci_get_error(ctx.get());
+        return false;
+    }
+
+    // If section does not exists, create new one, otherwise change the "type" of the section
+    if (uci_set(ctx.get(), &ptr) != UCI_OK) {
+        LOG(ERROR) << "Failed to add new section";
+        return false;
+    }
```

Comment:
Updated the function, it checks if the section already exists before calling the `uci_set`
according to the "if not exist add - otherwise, return false" logic